src/cli.ts

```diff
@@ -1,72 +1,117 @@
 #! /usr/bin/env node
-import { PlainObjectType } from './lib/types';
 import { ChameleonOptions } from './lib/interfaces';
 import * as chameleon  from './';
 import chalk from 'chalk';
 import yargs from 'yargs';
-import { getOptionsFromConfigFile } from './options';
-import { deepMerge } from './util';
+import  findUp from 'find-up';
+import * as fs from 'fs';
+import { getDefaultOptions, setupOptions } from './options';
+const configPath = findUp.sync(['chameleon.config.json', 'chameleon.config.js']);
 
-const { argv } = yargs
-.boolean(['css', 'scss', 'cApply','cPreserve', 'swApply', 'swNonScaling', 'tApply'])
-.string(['config', 'path', 'subdirName', 'fileName', 'cName', 'cCustomVars', 'swName', 'swCustomVars', 'tName', 'tDefault']);
-
-(async () => {
-  try {
-    const configOptions: Partial<ChameleonOptions> = await getOptionsFromConfigFile(argv.config);
-    const cliOptions: Partial<ChameleonOptions> = getOptionsAsObject();
-    const options: Partial<ChameleonOptions> = deepMerge<Partial<ChameleonOptions>>(configOptions, cliOptions);
-
-    await chameleon.create(options);
-  } catch(err) {
-    console.error(chalk.redBright(err));
-  }
-})();
-
-function getOptionsAsObject(): ChameleonOptions {
-  const colors = Object.assign({},
-      argv.cApply !== undefined && {apply: argv.cApply},
-    typeof argv.cName === 'string' && {name: argv.cName},
-    argv.cPreserve !== undefined && {preserveOriginal: argv.cPreserve},
-    typeof argv.cCustomVars === 'string' && {customVars: getCustomVarsAsObject(argv.cCustomVars)}
-  );
-
-  const strokeWidths = Object.assign({},
-    argv.swApply !== undefined && {apply: argv.swApply},
-    typeof argv.swName === 'string' && {name: argv.swName},
-    argv.swNonScaling !== undefined && {nonScaling: argv.swNonScaling},
-    typeof argv.swCustomVars === 'string' && {customVars: getCustomVarsAsObject(argv.swCustomVars)}
-  );
+const config = configPath ? JSON.parse(fs.readFileSync(configPath).toString()) : {};
 
-  const transition = Object.assign({},
-    argv.tApply !== undefined && {apply: argv.tApply},
-    typeof argv.tName === 'string' && {name: argv.tName},
-    typeof argv.tDefault === 'string' && {default: argv.tDefault},
-  );
+const { argv } = yargs
+    .option('css', {
```

Comment:
@christian-reichart we still might want to have a look at them now - i added them somewhat

Reply:
As I see right now, the dimensions styles shouldn't be the first options to list here, please keep the order of the options in the readme.
I see your worries about the descriptions, as they kind of break the layout inside the console because they are too long. maybe we need a shorter description for those, I can take a look